### PR TITLE
Harish v.next

### PR DESCRIPTION
**Pull request by harish-kunchala**
Created [show_wfs_layer_from_url](https://devtopia.esri.com/runtime/common-samples/tree/main/designs/AddWfsLayer) from this [issue](https://devtopia.esri.com/runtime/flutter/issues/2361). Ran the checks and made sure the app is working.

### CODE REVIEW

File: lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        loadFeatures();
+      }
+    });
+
+    await loadFeatures();
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  void onTap(Offset offset) {
```
[the remark this answers is not in the document]
Comment (harish-kunchala) [reply lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart]: Done. Removed it.

File: lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  }
+
+  Future<void> loadWfsLayerFromURL() async {
+    const wfsFeatureTableUri =
+        'https://dservices2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/services/Seattle_Downtown_Features/WFSServer?service=wfs&request=getcapabilities';
+
+    // Create a WFS feature table from URI and name.
+    _featureTable = WfsFeatureTable.withUriAndTableName(
+        uri: Uri.parse(wfsFeatureTableUri),
+        tableName: 'Seattle_Downtown_Features:Buildings')
```
[the remark this answers is not in the document]
Comment (harish-kunchala) [reply lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart]: Done. Added the Comma

File: lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
```
[the remark this answers is not in the document]
Comment (harish-kunchala) [reply lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart]: Removed the space line.

File: lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+      // Set the axis order and feature request mode.
+      ..axisOrder = OgcAxisOrder.noSwap
+      ..featureRequestMode = FeatureRequestMode.manualCache;
+
```
[the remark this answers is not in the document]
Comment (harish-kunchala) [reply lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart:113]: Removed the space line.

File: lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    // Get the current extent.
+    final currentExtent = _mapViewController.visibleArea;
+
+    // Create a query based on the current visible extent.
+    final visibleExtentQuery = QueryParameters()
+      ..geometry = currentExtent
+      ..spatialRelationship = SpatialRelationship.intersects;
+
+    try {
```
[the remark this answers is not in the document]
Comment (harish-kunchala) [reply lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart]: Removed the 'try/catch'.

File: lib/samples/show_wfs_layer_from_url/README.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+* ArcGISMapView
+* ArcGISMapViewController.onNavigationChanged
+* FeatureLayer
+* WfsFeatureTable
+* WfsFeatureTable.populateFromService
+
+## About the data
+
+This service shows building footprints for downtown Seattle. For additional information, see the underlying service on ArcGIS Online.
```
[the remark this answers is not in the document]
Comment (harish-kunchala) [reply lib/samples/show_wfs_layer_from_url/README.md]: Updated the README.md to show ArcGIS Online

File: lib/samples/show_wfs_layer_from_url/README.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+## How to use the sample
+
+Pan and zoom to see features within the current map extent.
+
+## How it works
+
+1. Create a `WfsFeatureTable` with a URL.
+2. Create a `FeatureLayer` from the feature table and add it to the map.
+3. Listen for the `ArcGISMapViewController.onNavigatonChanged` event to detect when the user has stopped navigating the map.
```
[the remark this answers is not in the document]
Comment (harish-kunchala) [reply lib/samples/show_wfs_layer_from_url/README.md]: Updated to say `when navigation is complete`.

File: lib/samples/show_wfs_layer_from_url/README.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+4. When the user is finished navigating, use `WfsFeatureTable.populateFromService` to load the table with data for the current visible extent.
```
[the remark this answers is not in the document]
Comment (harish-kunchala) [reply lib/samples/show_wfs_layer_from_url/README.md:20]: Updated to say 'when navigation is complete.`

**Comment by harish-kunchala** [comment]
Hey @changanxian. Thanks for your quick review, I've made the changes. Could you please check and let me know.

File: lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+      // Populate the table with the query, leaving existing table entries intact
+      await _featureTable.populateFromService(
+        parameters: visibleExtentQuery,
+        clearCache: false,
+        outFields: [],
+      );
+      // Hide the loading indicator
+      setState(() => _ready = true);
+
```
Comment (changanxian) [inline lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart:150]: extra space line

File: lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+    _featureTable = WfsFeatureTable.withUriAndTableName(
+        uri: Uri.parse(wfsFeatureTableUri),
+        tableName: 'Seattle_Downtown_Features:Buildings',)
+      // Set the axis order and feature request mode.
+      ..axisOrder = OgcAxisOrder.noSwap
+      ..featureRequestMode = FeatureRequestMode.manualCache;
+
+    // Create the feature layer from the feature table.
+    final featureLayer = FeatureLayer.withFeatureTable(_featureTable)
+
```
Comment (changanxian) [inline lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart:116]: extra space line

**Review by changanxian — APPROVED** [review]
otherwise, LGTM

File: lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+    _mapViewController.setViewpoint(initialViewPoint);
+
+    // Add a listener for map navigation events
+    _mapViewController.onNavigationChanged.listen((isNavigating) {
+      if (!isNavigating) {
+        loadFeatures();
+      }
+    });
+
+    await loadFeatures();
```
Comment (kmueller-gis) [inline lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart:97]: You may want to add a comment for this line saying that this is needed to load the features in the initial viewpoint

Comment (harish-kunchala) [reply lib/samples/show_wfs_layer_from_url/show_wfs_layer_from_url.dart:96]: Done added the comment for `loadFeatures()`

**Pull request merged**